operator/controllers/clusterconfig_controller.go

```diff
@@ -47,15 +58,72 @@ type ClusterConfigReconciler struct {
 // For more details, check Reconcile and its Result here:
 // - https://pkg.go.dev/sigs.k8s.io/controller-runtime@v0.12.1/pkg/reconcile
 func (r *ClusterConfigReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
-	_ = log.FromContext(ctx)
+	reqLogger := log.WithValues("Request.Name", req.Name)
+	reqLogger.Info("reconciling ClusterConfig: START")
+	defer reqLogger.Info("reconciling ClusterConfig: STOP")
+
+	instance := &metalk8sscalitycomv1alpha1.ClusterConfig{}
+	err := r.Get(ctx, req.NamespacedName, instance)
+	if err != nil {
+		if errors.IsNotFound(err) {
+			if req.Name == instanceName {
+				panic(fmt.Errorf(
+					"%s ClusterConfig object should not be deleted", req.Name,
+				))
+			}
+			reqLogger.Info("ClusterConfig already deleted: nothing to do")
+			return endReconciliation(nil)
+		}
+		reqLogger.Error(err, "cannot read ClusterConfig: requeue")
+		return delayedRequeue(err)
+	}
+
+	if instance.Name != instanceName {
+		if err := r.Delete(ctx, instance); err != nil {
+			reqLogger.Error(
+				err, "cannot delete extra ClusterConfig: requeue",
+			)
+			return delayedRequeue(err)
+		}
+		reqLogger.Info("deleting extra ClusterConfig, consider updating the default one")
```

Comment:
Would be better to print the actual `instanceName` in this message I believe

Reply:
Hmm yes indeed, I will fix it

Reply:
Done